Komodo class warship

spockye opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

This adds the Komodo-Class Warship, a decent size ship used by the syndicate (Gorlex marauders in particular) for the war effort. 
![image](https://user-images.githubusercontent.com/79304582/175380908-e2285cd1-86d0-4571-b13b-d58ddade20b2.png)

Lore wise, the ship used to be a top of the line warship though as the war ended, and the Gorlex marauders became unable to sustain their warship, they were left with the orders to keep the ship operational at any cost.

 This leaves you with its current state. Due to the lack of funding, drastic measures were taken, such as scrapping the front artillery bays for cash, stripping the mech for parts and only keeping the ship barely functional.

when playing this ship, you are not meant to spec into research, or gaining better gear. you are to follow your standing orders, and restore the ships honour and integrity. the crew of the ship would still be following old protocol, and are to uphold Syndicate standards. 

TODO list:

- [x] - fix the bloody custom jobs
- [x] - finish the ruin version of the ship
- [x] - potential balance changes

 

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
more roleplay oriented ships, with challanges = good!

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:
add: adds a new ship(Komodo-class Syndicate warship
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

spockye committed:
the ship file!

spockye committed:
adds a sec lathe/printer board

spockye committed:
adds the Json and hop DM

spockye committed:
attempts to fixy

spockye committed:
some more fixes

spockye committed:
adds custom jobs

oh god, i REALLY hope i did it right

spockye committed:
job DM fix

well, i fucked up

spockye committed:
final(final)final fix

lord forgive me, for i have sinned

spockye committed:
Update gorlex_Komodo.dmm

Apogee-dev commented on code/modules/jobs/job_types/security_officer.dm:
```diff
@@ -278,3 +278,13 @@ GLOBAL_LIST_INIT(available_depts, list(SEC_DEPT_ENGINEERING, SEC_DEPT_MEDICAL, S
 	keyslot = new /obj/item/encryptionkey/headset_sec
 	keyslot2 = new /obj/item/encryptionkey/headset_sci
+
+	/datum/outfit/syndicate_empty/sbc/assault_komodo
```
You accidentally nested this line underneath the previous block (starting with /obj/item/radio/headset/headset_sec/alt/department/sci). Delete a tab at the start of this line to fix the problem. You should also move this entire block up to before the "//Shiptest outfits end" comment in this file.

Apogee-dev replied:
Well, I meant to request changes, not approve, but still.

spockye replied:
That would potentially make the roles work?

Apogee-dev replied:
fixing the indentation absolutely will. Moving the code block up above the comment is just for the sake of organization.

spockye replied:
I'll have some time next sunday to get it fixed. If the bot marks this as stale somehow, im gonna hunt down its servers location and hurl insults at it

Latentish replied:
this was fixed

Valorium-RB commented:
Good looking ship!

Latentish commented:
I have no real particular complaints with how this is mapped. I really enjoy it and I think it's a prime example of ship design. However you happened to make a ship we basically already have. While I personally like this design more, the Gorlex Hyena is already a Syndicate assault ship. I would like this ship a lot more if you turned this into titanium hulls, nanotrasen weapons, and nanotrasen theming.

Edit: Regardless of theming, I am going to start asking all ships have a bathroom and running water somewhere.

Apogee-dev commented:
> I have no real particular complaints with how this is mapped. I really enjoy it and I think it's a prime example of ship design. However you happened to make a ship we basically already have. While I personally like this design more, the Gorlex Hyena is already a Syndicate assault ship. I would like this ship a lot more if you turned this into titanium hulls, nanotrasen weapons, and nanotrasen theming.
> 
> Edit: Regardless of theming, I am going to start asking all ships have a bathroom and running water somewhere.

The Hyena is a salvage ship, not an assault ship, which is why it gets so much mining gear and has no actual security positions. It got a bit of a reputation as a combat ship primarily due to nerfs to Syndie equipment coming in much slower than expected (or not at all yet, in the case of hardsuits and softsuits). It's also something like half the size of the Komodo, so I don't really consider the two comparable.

Latentish commented:
> The Hyena is a salvage ship, not an assault ship, which is why it gets so much mining gear and has no actual security positions. It got a bit of a reputation as a combat ship primarily due to nerfs to Syndie equipment coming in much slower than expected (or not at all yet, in the case of hardsuits and softsuits). It's also something like half the size of the Komodo, so I don't really consider the two comparable.

Yeah I suppose that's not a bad point. But I still feel like nanotrasen could use the dub more than syndicate. BUT it's extremely understandable that someone wouldn't want to redo it to this extent. It's a great ship still.

But I still require you to put a bathroom with a sink on the ship somewhere.

spockye committed:
Update security_officer.dm

spockye committed:
hopefylly(hopefully)possibly fixes jobs

spockye committed:
Merge branch 'master' into Komodo-Class-warship

spockye committed:
a oopsie woopsie

accidental enter

spockye committed:
adds missing pipe

Latentish reviewed (approved):
This is really one of the better ships I've seen added recently. I want to see this testmerged. You really nailed it with this one.

spockye committed:
moves a gun, changes some fuckery

Apogee-dev commented on code/modules/clothing/under/syndicate.dm:
```diff
@@ -272,3 +272,13 @@
 	var/obj/item/clothing/accessory/maidapron/inteq/A = new (src)
 	attach_accessory(A)
+
+/datum/outfit/syndicate/assult_komodo
```
delet this outfit, it's in the wrong place and a copy of the one you actually want to be using

Apogee-dev commented on _maps/configs/komodo.json:
```diff
@@ -0,0 +1,43 @@
+			"slots": 1
+		},
+		"Trooper":{
+			"outfit": "/datum/outfit/syndicate/assult_komodo",
```
change this path to /datum/outfit/syndicate_empty/assult_komodo, and also correct that to "assault_komodo" to fix the typo.

Latentish replied:
this was fixed

spockye committed:
fixes the massive uplink issues(hopefully)

anger

spockye committed:
fixes and adds door locks

also more balance changes

Derpguy3 commented:
The power monitoring console in engineering doesn't seem to be hooked up to the grid, and isn't showing any draw or supply and rooms on the list.

Latentish commented:
Please put a chair in front of the helm console.

Apogee-dev commented on code/modules/jobs/job_types/head_of_security.dm:
```diff
@@ -74,14 +74,14 @@
 	suit = /obj/item/clothing/suit/armor/vest/syndie
-	r_pocket = /obj/item/melee/transforming/energy/sword
-	backpack_contents = list(/obj/item/melee/baton/loaded=1)
+	id = /obj/item/card/id/syndicate_command/operative
```
replace this with /obj/item/card/id/syndicate_command/crew_id , the operative id doesn't actually display the character's real name. Additionally, make sure every job has this type of id if it doesn't inherit a syndie crew or captain id from its parent type.

spockye committed:
adds the correct id types, and fixes  some ickies

oh my

spockye committed:
misc changes

adds another mag for the cr-20
fixes button directions
adds some more misc clothing
fixes windoor directions
fixes power monitor
adds more extinguisher cabinets
some bodybags in the medic's locker

Latentish reviewed (approved):
Played this on live, didn't notice any issues. Going to marg this.

Apogee-dev reviewed (approved):
Forgot to approve this one. Looks great, has been working out pretty well in testing after you ironed out the initial issues.
